cli/metrics/metrics_build.go

```diff
@@ -0,0 +1,196 @@
+	if buildxConfig := os.Getenv("BUILDX_CONFIG"); buildxConfig != "" {
+		return buildxConfig
+	}
+	return filepath.Join(filepath.Dir(dockercfg.Filename), "buildx")
+}
+
+// buildxDriver returns the build driver being used for the build command
+func buildxDriver(dockercfg *configfile.ConfigFile, buildArgs []string) string {
+	driver := "error"
+	configDir := buildxConfigDir(dockercfg)
+	if _, err := os.Stat(configDir); err != nil {
+		return driver
+	}
+	builder := buildxBuilder(buildArgs)
+	if len(builder) == 0 {
+		// if builder not defined in command, seek current in buildx store
+		// `${DOCKER_CONFIG}/buildx/current`
+		fileCurrent := path.Join(configDir, "current")
+		if _, err := os.Stat(fileCurrent); err != nil {
+			return driver
+		}
+		// content looks like
+		// {
+		//   "Key": "unix:///var/run/docker.sock",
+		//   "Name": "builder",
+		//   "Global": false
+		// }
+		rawCurrent, err := ioutil.ReadFile(fileCurrent)
+		if err != nil {
+			return driver
+		}
+		// unmarshal and returns `Name`
+		var obj map[string]interface{}
+		if err = json.Unmarshal(rawCurrent, &obj); err != nil {
+			return driver
+		}
+		if n, ok := obj["Name"]; ok {
+			builder = n.(string)
+			// `Name` will be empty if `default` builder is used
+			// {
+			//   "Key": "unix:///var/run/docker.sock",
+			//   "Name": "",
+			//   "Global": false
+			// }
+			if len(builder) == 0 {
+				builder = "default"
+			}
+		} else {
+			return driver
+		}
+	}
+
+	// if default builder return docker
+	if builder == "default" {
+		return "docker"
+	}
+
+	// read builder info and retrieve the current driver
+	// `${DOCKER_CONFIG}/buildx/instances/<builder>`
+	fileBuilder := path.Join(configDir, "instances", builder)
+	if _, err := os.Stat(fileBuilder); err != nil {
+		return driver
+	}
+	// content looks like
+	// {
+	//   "Name": "builder",
+	//   "Driver": "docker-container",
+	//   "Nodes": [
+	//     {
+	//       "Name": "builder0",
+	//       "Endpoint": "unix:///var/run/docker.sock",
+	//       "Platforms": null,
+	//       "Flags": null,
+	//       "ConfigFile": "",
+	//       "DriverOpts": null
+	//     }
+	//   ],
+	//   "Dynamic": false
+	// }
+	rawBuilder, err := ioutil.ReadFile(fileBuilder)
+	if err != nil {
+		return driver
+	}
+	// unmarshal and returns `Driver`
+	var obj map[string]interface{}
+	if err = json.Unmarshal(rawBuilder, &obj); err != nil {
+		return driver
+	}
+	if d, ok := obj["Driver"]; ok {
+		driver = d.(string)
+	}
+	// TODO(@crazy-max): include buildkit version being used by this driver (e.g., docker-container;0.9.2)
+	return driver
+}
+
+// buildxBuilder returns the builder being used in the build command
+func buildxBuilder(buildArgs []string) string {
+	var builder string
+	fset := pflag.NewFlagSet("buildx", pflag.ContinueOnError)
+	fset.String("builder", "", "")
+	_ = fset.ParseAll(buildArgs, func(flag *pflag.Flag, value string) error {
+		if flag.Name == "builder" {
+			builder = value
+		}
+		return nil
+	})
+	if len(builder) == 0 {
```

Comment:
nit:
```suggestion
	if builder != "" {
```
?

Reply:
@mat007 I haven't reviewed the PR but the suggested logic is the opposite of the original logic

Reply:
Yes not the same logic also len is more efficient that string comp.

Reply:
There is no difference in the generated code, it’s just in my opinion easier to read with `== ""` as you know right away it’s a string.
See https://stackoverflow.com/a/18595217/1195832

Reply:
Yeah more a special taste then as they said. Back in the days `len(s) == 0` made sense.